### ISSUE TITLE
List all available ABI strings in reference.md

### ISSUE THREAD

**Issue by liigo** [rust-lang/rust#34267]

**Comment by nagisa** [comment]
Thing with these “ABI” string is that its dubious how many of them we want to consider stable. `C` is obvious as are microsoft’s ABIs… but everything else… *shrugs*?

**Comment by liigo** [comment]
```
extern "foo" {}

fn main() {
}
```
rustc may prints all available abi:
```
<anon>:1:8: 1:13 error: invalid ABI: expected one of [cdecl, stdcall, fastcall, vectorcall, aapcs, win64, Rust, C, system, rust-intrinsic, rust-call, platform-intrinsic], found `foo`
<anon>:1 extern "foo" {}
                ^~~~~
error: aborting due to previous error
```

**Comment by jethrogb** [comment]
For reference, the LLVM supported calling conventions are http://llvm.org/docs/doxygen/html/namespacellvm_1_1CallingConv.html and the rust calling conventions are https://github.com/rust-lang/rust/blob/c66d2380a810c9a2b3dbb4f93a830b101ee49cc2/src/libsyntax/abi.rs#L81-L100

**Comment by ubsan** [comment]
I think that "Rust" and "C" calling conventions should be declared "cross-compiler"; however, I think the other calling conventions should be compiler specific. These are the safest ABIs we can reasonably say are probably everywhere.

**Comment by jethrogb** [comment]
@ubsan Also "system"

**Comment by ubsan** [comment]
@jethrogb I don't know if "system" should be required by every compiler. "Rust" and "C" seem far more universal, whereas "system" only does anything on x32 Windows.

Might be enough code uses "system" to justify its inclusion, however.

**Comment by brson** [comment]
Removed the easy tag since there's no agreement about what to document.

**Comment by ubsan** [comment]
Fixed by #34461
